src/utils/utils.py

```diff
@@ -33,6 +33,90 @@
 _MOSES_DETOKENIZER = MosesDetokenizer()
 
 
+def check_for_previous_checkpoints(serialization_dir, tasks, phase, load_model):
+    """
+    Check if there are previous checkpoints. 
+    If phase == target_train, we loop through each of the tasks from last to first 
+    to find the task with the most recent checkpoint. 
+    If phase == pretrain, we check if there is a most recent checkpoint in the run 
+    directory. 
+
+    Parameters 
+    ---------------------
+    serialization_dir: str, 
+    tasks: List of SamplingMultiTask objects, 
+    phase: str 
+    load_model: bool
+
+    Returns
+    ---------------------
+    task_directory: None or str
+    max_epoch: int, -1 if not found
+    suffix: None or str, the suffix of the checkpoint
+    """
+    task_directory = None
+    suffix = None
+    ckpt_epoch = -1
+    ckpt_suffix = None
+    if phase == "target_train":
+        for task in tasks[::-1]:
+            max_epoch, suffix = find_last_checkpoint_epoch(serialization_dir, phase, task.name)
+            # if we haven't found a task directory yet
```

Comment:
Caps

Reply:
Why > -1? Seems to contradict the comment.

Reply:
Or I'm missing something.

Reply:
Oh this is a misleading comment - I'll clarify.